test.py

```diff
@@ -1776,19 +1796,24 @@ async def main() -> int:
     options = parse_cmd_line()
 
     open_log(options.tmpdir, f"test.py.{'-'.join(options.modes)}.log", options.log_level)
-
+    setup_cgroup(options.gather_metrics)
     await find_tests(options)
     if options.list_tests:
         print('\n'.join([f"{t.suite.mode:<8} {type(t.suite).__name__[:-9]:<11} {t.name}"
                          for t in TestSuite.all_tests()]))
         return 0
 
     signaled = asyncio.Event()
+    stop_event = asyncio.Event()
+    resource_watcher = run_resource_watcher(options.gather_metrics, signaled, stop_event, options.tmpdir)
 
     setup_signal_handlers(asyncio.get_running_loop(), signaled)
 
     try:
         await run_all_tests(signaled, options)
+        stop_event.set()
+        async with asyncio.timeout(5):
```

Comment:
Thanks. In Seastar we can't cancel things, so we don't have this pattern.

Reply:
But, since this is just a safety mechanism, I suggest to increase the timeout even more (since we're trying to load the machine), and report when it triggered.

Reply:
It's not cancelling, it's stopping of the monitoring. We are using this event as a flag that we need to stop the loop. I think this timeout is enough since it double the time of the sleep in waiting process. The only thing that I should implement is loggin an error in case of timeout happens, rather than failing the whole process. In the worst case, we will lose only the last metric with a CPU and RAM utilization, the rest should be already written.